### PR TITLE
Nex 72: Social share component

### PR DESCRIPTION
**Pull request by Zionknight-crypto**
[*Nex 72*](https://wunder.atlassian.net/browse/NEX-72)

*Changes proposed in this PR:*
- Create a social media share component

*How to test:*
1. Open the link and click on the share buttons to observe that they lead you to login into your social media account and a post should be available too.

2. Please note that there is an EmberJs console error for LinkedIn share link. Currently looking into it and contacting them. They have a js script plugin but would prefer not to use that.

*Link to feature environment:*
[Link](https://feature-nex-72-socialshare-next.next4drupal-project.dev.wdr.io/)

### CODE REVIEW

File: next/components/social-share.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <div>
+      <p className="text-center">{t("share-page")}</p>
+      <ul className="flex flex-wrap justify-center">
+        {data?.map((link) => (
+          <li className="m-4" key={link.id}>
+            <a href={link.url} target="_blank" rel="noopener noreferrer">
+              {link.icon}
+              <span className="sr-only">{`${t("share-to")} ${
+                link.location
+              }`}</span>
```
Comment (joshua-scott) [inline next/components/social-share.tsx:54]: `t` is being used a bit wrong here, according to your translation it should be something like `t("share-to", { location })`

File: next/components/social-share.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    setPageUrl(currentUrl);
+  }, [pageUrl]);
+
+  const data = [
+    {
+      id: 1,
+      location: "Facebook",
+      url: `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(
+        pageUrl
+      )}`,
```
Comment (joshua-scott) [inline next/components/social-share.tsx:23]: As a small cleanup, we could call `encodeURIComponent` just once when we calculate it?

File: next/components/social-share.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import TwitterIcon from "../styles/icons/twitter.svg";
+
+export function SocialShare() {
+  const [pageUrl, setPageUrl] = useState<string>("");
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    const currentUrl = window.location.href;
+    setPageUrl(currentUrl);
+  }, [pageUrl]);
```
Comment (joshua-scott) [inline next/components/social-share.tsx:15]: `pageUrl` is added to the dependency array, but I don't think it's needed, right?

File: next/components/social-share.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { useTranslation } from "next-i18next";
+import { useEffect, useState } from "react";
+
+import FacebookIcon from "../styles/icons/facebook.svg";
+import LinkedInIcon from "../styles/icons/linkedin.svg";
+import TwitterIcon from "../styles/icons/twitter.svg";
```
Comment (joshua-scott) [inline next/components/social-share.tsx:6]: Can use absolute imports like `"@/styles/icons/facebook.svg";` here

**Review by joshua-scott — COMMENTED** [review]
All icons are visible in Chrome, but strangely only one is visible in FF?

image

File: next/components/social-share.tsx (hunk 1 above)
Comment (Zionknight-crypto) [reply next/components/social-share.tsx]: Fixed now.

File: next/components/social-share.tsx (hunk 2 above)
Comment (Zionknight-crypto) [reply next/components/social-share.tsx]: Sorted.

File: next/components/social-share.tsx (hunk 4 above)
Comment (Zionknight-crypto) [reply next/components/social-share.tsx]: fixed.

**Review by joshua-scott — APPROVED** [review]

**Pull request merged**